### PR TITLE
condensed port forwarding calls

### PR DESCRIPTION
**Pull request by kdrumm**

### CODE REVIEW

File: modules/virtue-admin/src/main/java/com/ncc/savior/virtueadmin/infrastructure/mixed/XenGuestManager.java [hunk 1]
```diff
@@ -288,15 +288,20 @@ private void setupPortForwarding(Session session, int externalSensingPort, int s
...
+//		for (int i = 0; i < numSensingPorts; i++) {
+//			dataModel.put("externalPort", externalSensingPort + i);
+//			dataModel.put("internalPort", startingInternalPort + i);
+//			List<String> o = SshUtil.runScriptFromFile(templateService, session, "port-forwarding.tpl", dataModel);
+//			lines.addAll(o);
+//			JavaUtil.sleepAndLogInterruption(100);
+//		}
 		JavaUtil.sleepAndLogInterruption(100);
-		logger.debug(lines.toString());
+		logger.debug(lines.toString() );
```
Comment (chrislong) [inline modules/virtue-admin/src/main/java/com/ncc/savior/virtueadmin/infrastructure/mixed/XenGuestManager.java]: extra space

File: modules/virtue-admin/src/main/resources/templates/port-forwarding.tpl [hunk 2]
```diff
@@ -15,5 +18,10 @@ portForward () {
...
 }
 
-
-portForward ${externalPort} ${guestIp} ${internalPort}
+for (( i=0; i<$NUM_PORTS; i++ ))
+do
+	extPort=$((i + ${externalPort}))
+	intPort=$((i + ${internalPort}))
+	echo "i=$i Port $extPort -> $intPort"
+	portForward $extPort ${guestIp} $intPort
```
Comment (chrislong) [inline modules/virtue-admin/src/main/resources/templates/port-forwarding.tpl]: I was curious so I looked in the script. You *could* collect the output from the portForward calls for the REMOTEPORTFILE and do one ssh command to create that file. I'm not sure if it's worth it, though. What's that file for, anyway?

Comment (kdrumm) [reply modules/virtue-admin/src/main/resources/templates/port-forwarding.tpl]: That is the file that the sensors use to know what host and port they are from.

File: modules/virtue-admin/src/main/java/com/ncc/savior/virtueadmin/infrastructure/mixed/XenGuestManager.java (hunk 1 above)
Comment (kdrumm) [reply modules/virtue-admin/src/main/java/com/ncc/savior/virtueadmin/infrastructure/mixed/XenGuestManager.java]: fixed

**Comment by kdrumm** [comment]
Checking the timing, the vast majority of the time is from these 4 tasks:
- Startup of dom0
- Startup of Xen (waiting for xl list)
- Startup of dom U
- Startup of Xpra

The first 2 should be solved from the xen pools.  The 3rd and 4th I don't know how to solve.  Do you have any ideas on how to speed up linux boot?

**Comment by chrislong** [comment]
Ok to merge

**Pull request merged**